ops/terraform/services/server/server-regression/iam.tf

```
@@ -326,3 +326,86 @@ resource "aws_iam_role" "glue_trigger" {
     aws_iam_policy.glue.arn
   ]
 }
+
+resource "aws_iam_role" "spice_refresh_role" {
+  name = local.spice_trigger_lambda_name
+  assume_role_policy = jsonencode(
+    {
+      Statement = [
+        {
+          Action = "sts:AssumeRole"
+          Effect = "Allow"
+          Principal = {
+            Service = "lambda.amazonaws.com"
+          }
+        },
+      ]
+      Version = "2012-10-17"
+    }
+  )
+  force_detach_policies = true
+}
+
+resource "aws_iam_policy" "spice_refresh_logs_policy" {
+  name = "${local.spice_trigger_lambda_name}-logs"
+  description = join("", [
+    "Permissions for the ${local.spice_trigger_lambda_name} Lambda to write to its corresponding ",
+    "CloudWatch Log Group and Log Stream",
+  ])
+  policy = jsonencode(
+    {
+      Statement = [
+        {
+          Action   = "logs:CreateLogGroup"
+          Effect   = "Allow"
+          Resource = "arn:aws:logs:${local.region}:${local.account_id}:*"
+        },
+        {
+          Action = [
+            "logs:CreateLogStream",
+            "logs:PutLogEvents",
+          ]
+          Effect = "Allow"
+          Resource = [
+            "arn:aws:logs:${local.region}:${local.account_id}:log-group:/aws/lambda/${local.spice_trigger_lambda_name}:*",
+          ]
+        },
+      ]
+      Version = "2012-10-17"
+    }
+  )
+}
+
+resource "aws_iam_policy" "spice_refresh_policy" {
+  name = "${local.spice_trigger_lambda_name}-quicksight"
+  description = join("", [
+    "Policy that allows searching and refreshing SPICE data sets for the ",
+    "bfd-${local.env}-${local.service} refresh spice trigegr lambda",
```

```suggestion
    "{local.spice_trigger_lambda_name} Lambda",
```

Looks like this is a spot where the `spice_trigger_lambda_name` local could be used in-place of manually constructing the name.